pdr_backend/accuracy/app.py

```diff
@@ -0,0 +1,94 @@
+import threading
```

Comment:
General comment for every new function added in this PR:

Each new function needs:
- a docstring
- one or more unit tests
- a `@enforce_types` decorator on top (so that dynamic type checking is actually invoked)

This is extremely important to maintainability and usability.

Reply:
- Docs are added to functions
- @enforce_types are added and tested
- Unit tests are not ready yet

Reply:
Looks like docstrings are there, great.

Looks like some @enforce_types are still missing.